runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilitySession.java

```diff
@@ -53,8 +53,15 @@ void incrementLapCount(Client client, XpTrackerService xpTrackerService)
 		final int currentExp = client.getSkillExperience(Skill.AGILITY);
 		final int goalXp = xpTrackerService.getEndGoalXp(Skill.AGILITY);
 		final int goalRemainingXp = goalXp - currentExp;
+		double courseTotalExp = course.getTotalXp();
+		if (course == Courses.PYRAMID)
+		{
+			// agility pyramid has a bonus exp drop on the last obstacle that scales with player level and caps at 1000
+			// the bonus is not already accounted for in the total exp number in the courses enum
```

Comment:
This line is inaccurate. `totalXp` includes all experience gained during a lap of the course, including the final obstacle.

Reply:
that is what it should be, but for the agility pyramid it does not, and the reason the comment is there at all is to explain that and why the code under it is doing what its doing.

Reply:
Given that the pyramid course already checks for a lap completion specifically by the ending location, I would rather see its `Course` definition changed so the ending obstacle xp can be fully replaced here instead.